app.js

```diff
@@ -0,0 +1,78 @@
+var express = require('express');
+var path = require('path');
+var favicon = require('serve-favicon');
+var logger = require('morgan');
+var cookieParser = require('cookie-parser');
+var bodyParser = require('body-parser');
+var connect = require('connect');
+
+var routes = require('./routes/index');
+var users = require('./routes/users');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'ejs');
+
+// uncomment after placing your favicon in /public
+//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
+// Existing code
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
+// Method override – NEW CODE
+var connect        = require('connect')
+var methodOverride = require('method-override')
+app.use(methodOverride(function(req, res){
+  if (req.body && typeof req.body === 'object' && '_method' in req.body) {
+    // look in urlencoded POST bodies and delete it
+    var method = req.body._method
+    delete req.body._method
+    return method
+  }
+}))
+
+// Mongoose connection – NEW CODE
+var mongoose = require('mongoose');
+mongoose.connect(process.env.DB_GA_STUDENTS);
+
+// Existing code
+app.use('/', routes);
```

Comment:
Remember, according to the prompt you want all these routes to be prefixed with `/pirates`, so you know what entity you're working with in the URL.